Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -835,4 +858,12 @@ ComplexTextController::ComplexTextRun::ComplexTextRun(const Vector<FloatSize>& a
 {
 }
 
+float ComplexTextController::measuredTextSpacing(const Font& font)
+{
+    if (!m_measuredTextSpacing)
+        m_measuredTextSpacing =  0.125 * font.fontMetrics().ideogramWidth().value_or(0);
+
+    return m_measuredTextSpacing.value();
+}
```

Comment:
If the font is something that changes over the course of the run, then this is going to return the wrong results. And if it's not, it makes more sense to just compute it up front and leave it there rather than doing this kind of optional caching, no?

Reply:
The font shouldn't change within a `ComplexTextRun`, but you are right, we should compute it up front on the creation of `ComplexTextRun`.